### PR TITLE
Feature/ecs resource fleet

### PR DESCRIPTION
**Pull request by juhuahaoteng**
fleet_test

### CODE REVIEW

**Comment by juhuahaoteng** [comment]
> fleet_test

IT-C1MRWWDYH3QD:terraform-provider-alicloud chore$ TF_ACC=1 go test ./alicloud -v -run=TestAccAlicloudAutoProvisioningGroup  -timeout=300m
=== RUN   TestAccAlicloudAutoProvisioningGroup
--- SKIP: TestAccAlicloudAutoProvisioningGroup (2.26s)
    provider_test.go:246: Skipping the test case with there is no default vswitche
PASS
ok      github.com/terraform-providers/terraform-provider-alicloud/alicloud     2.360s

File: alicloud/resource_alicloud_ecs_autoprovisioning_group.go [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+			"spot_allocation_strategy": {
+				Type:         schema.TypeString,
+				Default:      "lowest-price",
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"lowest-price", "diversified"}, false),
+			},
+			"spot_instance_interruption_behavior": {
+				Type:     schema.TypeString,
+				Default:  "stop",
+				Optional: true,
```
Comment (Lexsss) [inline alicloud/resource_alicloud_ecs_autoprovisioning_group.go]: ValidateFunc for all fields

File: alicloud/resource_alicloud_ecs_autoprovisioning_group.go [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+		})
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+		response, _ := raw.(*ecs.CreateAutoProvisioningGroupResponse)
+		d.SetId(response.AutoProvisioningGroupId)
+		return nil
+	}); err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_ecs_autoProvisioning_group", request.GetActionName(), AlibabaCloudSdkGoERROR)
```
Comment (Lexsss) [inline alicloud/resource_alicloud_ecs_autoprovisioning_group.go]: alicloud_ecs_auto_provisioning_group

File: alicloud/resource_alicloud_ecs_autoprovisioning_group.go [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+func resourceAlicloudEcsAutoProvisioningGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*connectivity.AliyunClient)
+	request := ecs.CreateModifyAutoProvisioningGroupRequest()
+	request.RegionId = client.RegionId
+	request.AutoProvisioningGroupId = d.Id()
+
+	d.Partial(true)
+	if d.HasChange("excess_capacity_termination_policy") {
+		if v := d.Get("excessCapacity_termination_policy"); v != nil {
```
Comment (Lexsss) [inline alicloud/resource_alicloud_ecs_autoprovisioning_group.go]: d.GetOk

File: alicloud/resource_alicloud_ecs_autoprovisioning_group_test.go [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func TestAccAlicloudAutoProvisioningGroup(t *testing.T) {
+	var v ecs.AutoProvisioningGroup
+	rand := acctest.RandIntRange(10000, 999999)
+	name := fmt.Sprintf("tf-testAccautoprovisioninggroup-%d", rand)
+	var basicMap = map[string]string{
+		"auto_provisioning_group_name":               "ddd",
```
Comment (Lexsss) [inline alicloud/resource_alicloud_ecs_autoprovisioning_group_test.go]: use TestAcc as prefix

File: alicloud/resource_alicloud_ecs_autoprovisioning_group_test.go [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+				Config: testAccConfig(map[string]interface{}{
+					"auto_provisioning_group_name": "sss",
+				}),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheck(map[string]string{
+						"auto_provisioning_group_name": "sss",
+					}),
+				),
+			},
+		},
```
Comment (Lexsss) [inline alicloud/resource_alicloud_ecs_autoprovisioning_group_test.go]: add a step for modify all fields

File: alicloud/resource_alicloud_ecs_autoprovisioning_group_test.go [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+          instance_type                 = "ecs.n1.tiny"
+          security_group_id             = "${alicloud_security_group.default.id}"
+    }
+    data "alicloud_vswitches" "default" {
+		  zone_id = "${data.alicloud_zones.default.ids[0]}"
+          name_regex = "default-tf-testAcc-00"
+    }`, EcsInstanceCommonTestCase, name)
+}
+
+func testAccCheckEcsAutoProvisioningGroupDestroy(s *terraform.State) error {
```
Comment (Lexsss) [inline alicloud/resource_alicloud_ecs_autoprovisioning_group_test.go]: I don't think this is necessary

File: alicloud/resource_alicloud_ecs_autoprovisioning_group.go [hunk 7]
```diff
@@ -0,0 +1,394 @@
...
+
+	if err != nil {
+		if IsExpectedErrors(err, []string{"InvalidAutoProvisioningGroupId.NotFound"}) {
+			return nil
+		}
+		return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+	}
+	addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+
+	return WrapError(ecsService.WaitForEcsAutoProvisioningGroup(d.Id(), DefaultTimeout))
```
Comment (Lexsss) [inline alicloud/resource_alicloud_ecs_autoprovisioning_group.go]: same method and parameters with create func?

File: website/docs/r/ecs_autoprovisioning_group.html.markdown [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+sidebar_current: "docs-alicloud-resource-autoprovisioning-group"
+description: |-
+  Provides a ECS AutoProvisioning group resource.
+---
+
+# alicloud\_ecs\_autoprovisioning\_group
+
+Provides a ECS autoProvisioning group resource which is a solution that uses preemptive instances and pay_as_you_go instances to rapidly deploy clusters.
+
+-> **NOTE:** Available in 1.76.0+
```
Comment (Lexsss) [inline website/docs/r/ecs_autoprovisioning_group.html.markdown]: 1.77.0

File: alicloud/resource_alicloud_ecs_autoprovisioning_group.go (hunk 7 above)
Comment (juhuahaoteng) [reply alicloud/resource_alicloud_ecs_autoprovisioning_group.go]: create and then describe groups, OK?

File: alicloud/resource_alicloud_ecs_autoprovisioning_group.go (hunk 3 above)
Comment (juhuahaoteng) [reply alicloud/resource_alicloud_ecs_autoprovisioning_group.go]: i don't understand.

File: alicloud/resource_alicloud_ecs_autoprovisioning_group.go [hunk 9]
```diff
@@ -0,0 +1,398 @@
...
+}
+
+func resourceAlicloudEcsAutoProvisioningGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(*connectivity.AliyunClient)
+	request := ecs.CreateModifyAutoProvisioningGroupRequest()
+	request.RegionId = client.RegionId
+	request.AutoProvisioningGroupId = d.Id()
+
+	d.Partial(true)
```
Comment (Lexsss) [inline alicloud/resource_alicloud_ecs_autoprovisioning_group.go]: I don't think the `Partial `  is necessary

File: website/docs/r/ecs_autoprovisioning_group.html.markdown [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+You can use to the existing [autoProvisioning module](https://registry.terraform.io/modules/terraform-alicloud-modules/autoProvisioning/alicloud) 
+to create a autoProvisioning group.
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `launch_template_id` - (Required) The ID of the instance launch template associated with the auto provisioning group.
+* `total_target_capacity` - (Required) The total target capacity of the auto provisioning group. The target capacity consists of the following three parts:PayAsYouGoTargetCapacity,SpotTargetCapacity and the supplemental capacity besides PayAsYouGoTargetCapacity and SpotTargetCapacity.
+* `autoProvisioning_group_name` - (Optional) The name of the auto provisioning group to be created. It must be 2 to 128 characters in length. It must start with a letter but cannot start with http:// or https://. It can contain letters, digits, colons (:), underscores (_), and hyphens (-)
```
Comment (Lexsss) [inline website/docs/r/ecs_autoprovisioning_group.html.markdown]: auto_provisioning_group_name

File: website/docs/r/ecs_autoprovisioning_group.html.markdown [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+* `autoProvisioning_group_type` - (Optional) The type of the auto provisioning group. Valid values:`request` and `maintain`,Default value: `maintain`.
```
Comment (Lexsss) [inline website/docs/r/ecs_autoprovisioning_group.html.markdown]: still

File: website/docs/r/ecs_autoprovisioning_group.html.markdown [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+* `spot_allocation_strategy` - (Optional) The scale-out policy for preemptible instances. Valid values:`lowest-price` and `diversified`,Default value: `lowest-price`.
+* `spot_target_capacity` - (Optional) The target capacity of preemptible instances in the auto provisioning group.
+* `spot_instance_interruption_behavior` - (Optional) The default behavior after preemptible instances are shut down. Value values: `stop` and `terminate`,Default value: `stop`.
+* `spot_instance_pools_to_use_count` - (Optional) This parameter takes effect when the `SpotAllocationStrategy` parameter is set to `lowest-price`. The auto provisioning group selects instance types of the lowest cost to create instances.
+* `pay_as_you_go_allocation_strategy` - (Optional) The scale-out policy for pay-as-you-go instances. Valid values: `lowest-price` and `prioritized`,Default value: `lowest-price`.
+* `pay_as_you_go_target_capacity` - (Optional) The target capacity of pay-as-you-go instances in the auto provisioning group.
```
Comment (Lexsss) [inline website/docs/r/ecs_autoprovisioning_group.html.markdown]: default and validation description

File: alicloud/resource_alicloud_auto_provisioning_group.go [hunk 13]
```diff
@@ -0,0 +1,385 @@
...
+	}
+	if v, ok := d.GetOk("description"); ok && v.(string) != "" {
+		request.Description = v.(string)
+	}
+	if configs, ok := d.GetOk("launch_template_config"); ok {
+		confs := configs.([]interface{})
+		createConfigs := make([]ecs.CreateAutoProvisioningGroupLaunchTemplateConfig, 0, len(confs))
+		for _, c := range confs {
+			cc := c.(map[string]interface{})
+			conf := ecs.CreateAutoProvisioningGroupLaunchTemplateConfig{
```
Comment (Lexsss) [inline alicloud/resource_alicloud_auto_provisioning_group.go:372]: Check If the value exists

File: alicloud/resource_alicloud_auto_provisioning_group.go [hunk 14]
```diff
@@ -0,0 +1,385 @@
...
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"instance_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"max_price": {
+							Type:     schema.TypeString,
+							Required: true,
```
Comment (Lexsss) [inline alicloud/resource_alicloud_auto_provisioning_group.go:127]: computed and add to testcase

File: alicloud/resource_alicloud_auto_provisioning_group.go [hunk 15]
```diff
@@ -0,0 +1,385 @@
...
+						"vswitch_id": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"weighted_capacity": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"priority": {
+							Type:     schema.TypeString,
```
Comment (Lexsss) [inline alicloud/resource_alicloud_auto_provisioning_group.go:138]: still

File: website/docs/r/auto_provisioning_group.html.markdown [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+sidebar_current: "docs-alicloud-resource-auto-provisioning-group"
+description: |-
+  Provides a ECS Auto Provisioning group resource.
+---
+
+# alicloud\_auto\_provisioning\_group
+
+Provides a ECS auto provisioning group resource which is a solution that uses preemptive instances and pay_as_you_go instances to rapidly deploy clusters.
+
+-> **NOTE:** Available in 1.78.0+
```
Comment (Lexsss) [inline website/docs/r/auto_provisioning_group.html.markdown:14]: 1.79.0+

**Pull request merged**